### PR TITLE
Updating Cilium to 1.15.2

### PR DESCRIPTION
**Pull request by berkayoz**
Updated the helm chart to `1.15.2`
Updated cilium cli in e2e tests to `1.16.3`
Added `disableEnvoyVersionCheck: true` to ignore envoy version due to how we build rocks
Updated helm chart for LB to match [updated](https://docs.cilium.io/en/v1.15/network/lb-ipam/#pools) `CiliumLoadBalancerIPPool` CRD
Updated GatewayAPI to v1.0.0

### CODE REVIEW

**Review by bschimke95 — COMMENTED** [review]
Looks good. Minor comment. 

@neoaggelos we can close [this PR](https://github.com/canonical/k8s-snap/pull/220) then probably.

File: tests/integration/tests/test_gateway.py [hunk 1]
```diff
@@ -91,8 +91,6 @@ def test_gateway(instances: List[harness.Instance]):
     )
     gateway_http_port = p.stdout.decode().replace("'", "")
 
-    p = instance.exec(
-        ["curl", f"localhost:{gateway_http_port}"],
-        capture_output=True,
-    )
-    assert "Welcome to nginx!" in p.stdout.decode()
+    util.stubbornly(retries=5, delay_s=5).on(instance).until(
```
[the remark this answers is not in the document]
Comment (berkayoz) [reply tests/integration/tests/test_gateway.py:94]: It's just to make CI more resilient, I've run into some flakes while testing mainly due to ingress not being ready when we make the request(bit of a race condition here).

[the remark this answers is not in the document]
Comment (bschimke95) [reply tests/integration/tests/test_gateway.py:94]: Nice - I think I've also seen this quite some times already. Great that this is fixed now.

**Review by bschimke95 — APPROVED** [review]
LGTM

**Review by neoaggelos — APPROVED** [review]

**Pull request merged**